eddy/src/decryption_share.rs

```diff
@@ -2,18 +2,12 @@ use rand_core::{CryptoRng, RngCore};
 
 use crate::{limb, Ciphertext, PrivateKeyShare, PublicKeyShare, TranscriptProtocol};
 
-#[derive(Debug, Clone)]
-pub enum Verified {}
-#[derive(Debug, Clone)]
-pub enum Unverified {}
-
-pub trait VerificationState {}
-impl VerificationState for Verified {}
-impl VerificationState for Unverified {}
+pub struct Verified;
+pub struct Unverified;
```

Comment:
These types don't have any connection to each other, other than both being used as type parameters for `DecryptionShare<S>`, and there's no constraint on the `DecryptionShare` type parameter that it be `Verified` or `Unverified`. This means that someone could construct a `DecryptionShare<u64>` or whatever else, and it's also a bit confusing, because looking at the `DecryptionShare` there's no real clue about what the parameter is supposed to represent.

To improve this, the `Verified` and `Unverified` structs could implement a `VerificationMarker` trait (naming: it could also be called `VerificationState`, but calling it `Marker` suggests that it's only a marker type, rather than any runtime state).  The `VerificationMarker` trait could use the [sealed trait pattern](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed) to prevent downstream implementations.

There's an example of this in the `decaf377-rdsa` crate: https://rustdoc.penumbra.zone/main/decaf377_rdsa/trait.Domain.html